anxcloud/schema_virtual_server.go

```diff
@@ -18,17 +18,29 @@ func schemaVirtualServer() map[string]*schema.Schema {
 			ForceNew:    true,
 			Description: "Location identifier.",
 		},
+		"template": {
+			Type:         schema.TypeString,
+			ForceNew:     true,
+			Optional:     true,
+			ExactlyOneOf: []string{"template_id", "template"},
+			Description: "Named template. Can be used instead of the template_id to select a template. " +
+				"Format: `<name>[:(<build_no>|latest)]`. Omitting the build identifier is a short form of :latest. " +
+				"Example: (`Debian 11`, `Debian 11:latest`, `Debian 11:b42`).",
+		},
 		"template_id": {
-			Type:        schema.TypeString,
-			Required:    true,
-			ForceNew:    true,
-			Description: "Template identifier.",
+			Type:         schema.TypeString,
+			Optional:     true,
+			Computed:     true,
```

Comment:
do we need something like `computed = template != ""`, like the `customdiff.ForceNewIf` stuff above?

Reply:
Should be correct as is.